src/main/java/reposense/model/Author.java

```diff
@@ -5,30 +5,103 @@
 import java.util.ArrayList;
 import java.util.List;
 
+/**
+ * Represents a Git Author.
+ */
 public class Author {
+
     public static final String UNKNOWN_AUTHOR_GIT_ID = "-";
 
-    private String gitId;
+    private static final String MESSAGE_ILLEGAL_GIT_ID = "The provided Git ID, %s, contains illegal characters.";
+    private static final String MESSAGE_ILLEGAL_DISPLAY_NAME =
+            "The provided display name, %s, contains illegal characters.";
+    private static final String MESSAGE_ILLEGAL_AUTHOR_ALIAS =
+            "The provided author alias, %s, contains illegal characters.";
+    private static final String MESSAGE_UNCOMMON_GLOB_PATTERN =
+            "The provided ignore glob, %s, uses uncommon pattern.";
+
+    private static final String COMMON_GLOB_REGEX = "^[-a-zA-Z0-9 _/\\\\*!{}\\[\\]!(),:.]*$";
+    private static final String NAME_VALIDATION_REGEX = "^[-a-zA-Z0-9 _/\\\\*]+$";
```

Comment:
shouldn't this 2 be dump inside the function itself? i mean it works, but...

Reply:
For magic values, we usually represent them with a constant
- easier to maintain
- more readable with a meaningful name